configure.ac

```diff
@@ -1,18 +1,48 @@
 
 AC_PREREQ([2.69])
-AC_INIT([ledmon], [0.96])
+
+
+m4_define([MAJOR_VERSION], 0)
+m4_define([MINOR_VERSION], 97)
+m4_define([MICRO_VERSION], 0)
+
+AC_INIT([ledmon],[MAJOR_VERSION.MINOR_VERSION.MICRO_VERSION])
 AC_CONFIG_MACRO_DIR([m4])
 AC_SUBST([PACKAGE_DATE], "May 2022")
-AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+AM_INIT_AUTOMAKE([-Wall -Werror foreign subdir-objects])
+
+LT_PREREQ([2.4.6])
+AM_PROG_AR
+LT_INIT
 
 # Checks for programs.
 
 AX_CHECK_PROG([gcc])
 AX_CHECK_PROG([make])
 
-AM_PROG_CC_C_O
-AC_PROG_CC_C99
+
+# Using the version parts we will construct the appropriate libtool version.
```

Comment:
You mentioned here only library versioning, we wouldn't want to separate ledmon/ledctl versioning from library, because it's still one product from our side,
so changes in ledmon/ledct will also bump up the version

Reply:
I've change the `configure.ac` so there are two different versions available, 1 for binaries and 1 for library.

Reply:
Hello @tasleson ,

Having two separate versioning systems is overhead. For the maintenance reasons I need to have release model compatible across repository. I would prefer to stay with "X.Y" when X is as you proposed " when we have an incompatible library version" and Y is as now, number driver by needs, so bumping this number could be a result of:
	- New library function
	- New functionality in an application hosted here
	- Bunch of bugfixes
	- Business need for release

So far I know it is not common to have 2 independent release versions across one repository.
I'm fine with release overhead in some cases (e.g. when only library is changed or only ledmon).
 
We are going to release one product and later vendors will build they own packages basing on that.  The new release is a trigger of the action. They can build ledmon and libled and they both will have the same version. That is simple and consistent. One tag is a release. It is resolved same way in systemd (see that some libs are provided in the main systemd repository).

I'm totally against the "X.Y.Z" model where Z is bumped with every bugfix. The main reason comes also from maintenance purposes, every vendor package is named in following schema `<name>-<release>-<vendor_revision>`. 
As far I know, `<release>` represents particular tag in repository and that is the problem because tagging every bugfix is overhead and maintenance hell. I understand that library is updated rarely but I cannot guarantee that at the begging of the product. Having something like that will cause that libled-1.3.2 in Redhat and libled-1.3.2 in Suse could be based on different codebase (because micro .2 is not tagged). I would like to have customizations only with `<vendor_revision>` number.

If I'm wrong here and we will need to change versioning approach then it should be safe (adding .Z is safer than removing it, any package manager should be able to handle that properly).

Library is a kind of new start so I would like to bump version to "1.0" but as usual, it will be done in separate release patch, not now. After the merge I would like to left it unreleased for a while to give users a chance to reports bugs.
Does it work for you?

I will review the code once again soon, but don't worry, I'm not going to block that and request changes. I believe that we are good now but I still need take a look into changes after rebase.
We are waiting for validation team results now.

Let me know how you see it. I think that you tried to follow library BKM's so please let me know if there are other motivations I should take into account and use separate versioning.

Thanks,
Mariusz

Reply:
> Hello @tasleson ,
> 
> Having two separate versioning systems is overhead. For the maintenance reasons I need to have release model compatible across repository. I would prefer to stay with "X.Y" when X is as you proposed " when we have an incompatible library version" and Y is as now, number driver by needs, so bumping this number could be a result of: - New library function - New functionality in an application hosted here - Bunch of bugfixes - Business need for release

I'm following the industry expectation of https://semver.org/ 

> So far I know it is not common to have 2 independent release versions across one repository. I'm fine with release overhead in some cases (e.g. when only library is changed or only ledmon).

There is quite a bit of code that is different between the library and the command line utilities, I thought it would be good to split it up, to allow independent updates, but I can see the concern and the fact that the package version will need to get updated when either is changed.  It is likely better to keep 1 version.  Please note that the library version is used to drive the shared object version, eg. `libled.so.0.1.0` nothing else.

> We are going to release one product and later vendors will build they own packages basing on that. The new release is a trigger of the action. They can build ledmon and libled and they both will have the same version. That is simple and consistent. One tag is a release. It is resolved same way in systemd (see that some libs are provided in the main systemd repository).
> 
> I'm totally against the "X.Y.Z" model where Z is bumped with every bugfix. The main reason comes also from maintenance purposes, every vendor package is named in following schema `<name>-<release>-<vendor_revision>`. As far I know, `<release>` represents particular tag in repository and that is the problem because tagging every bugfix is overhead and maintenance hell. I understand that library is updated rarely but I cannot guarantee that at the begging of the product. Having something like that will cause that libled-1.3.2 in Redhat and libled-1.3.2 in Suse could be based on different codebase (because micro .2 is not tagged). I would like to have customizations only with `<vendor_revision>` number.

Any release of the project needs a source control tag for reproducibility.  There is nothing requiring Z being bumped for every bugfix.  It is simply bumped when 1 or more fixes have been made and a release has been made.  All distributions handle packaging versions the same way AFAIK in that if the upstream version matches then the package release is based on the same version.  However, the packaged version can contain it's own patches to the upstream source.

> 
> If I'm wrong here and we will need to change versioning approach then it should be safe (adding .Z is safer than removing it, any package manager should be able to handle that properly).
> 
> Library is a kind of new start so I would like to bump version to "1.0" but as usual, it will be done in separate release patch, not now. After the merge I would like to left it unreleased for a while to give users a chance to reports bugs. Does it work for you?

Referencing the above specification on semantic versioning this is handled with having a version where X == 0.

> I will review the code once again soon, but don't worry, I'm not going to block that and request changes. I believe that we are good now but I still need take a look into changes after rebase. We are waiting for validation team results now.
> 
> Let me know how you see it. I think that you tried to follow library BKM's so please let me know if there are other motivations I should take into account and use separate versioning.

I think we should go back to 1 version, taking existing version and adding a Z, eg. `0.97.0`.   At some point in the future we can bump X to 1 when the API is stable.  Because the version is quite important to users of the library we would need to be careful to only update Y when the API has an addition.  Thus the vast majority of updates will drive Z to get updated.

> 
> Thanks, Mariusz

Reply:
Thanks for clarification. I like better approach with 3 separate numbers so If we can make it this way:
```
MAJOR=0
MINOR=97
MICRO=0
```
that will fit better for me. But both way are correct so I have no objections for going back if you see that as not possible or overcomplicated.
Please also made the clear statement that we don't need to update Z with every bugfix because it already confused me. Feel free to add link to https://semver.org/ where it is well described.

I'm fine with 3 digits versioning style now. 

Thanks!
Mariusz

Reply:
Reverted back to what I had before, added URL and updated comments to be clear about not needing to increment release version for every single bug fix.